### ISSUE TITLE
state.json overwritten with new trial start

### ISSUE THREAD

**Issue by carolinedav** [kassonlab/run_brer#24]
`state.json` gives an overview of the BRER run for a given member. When a member restarts BRER with a new trial (ex. mem_1/0 > mem_1/1), the state file is overwritten with the new trial's information. It would be nice if we could maintain the run state for every trial (mainly for readability).

**Comment by eirrgang** [comment]
Can you clarify what information is not being retained in a usable form?

The proposed change uses a `state_{iteration}.json` file. It is not clear to me that the logic is right, but I also don't think this is quite the right way to go.

I don't think we want to conflate "current state" with "results". I expect that what we want is an additional file that serves a clear purpose distinct from the state file.

Note: We could retain more information in the single `state.json` file, but I am disinclined to do that. I wouldn't see any conceptual conflict if we decide to use `state.json` to store the state for all "iterations" in addition to the parameter of "current iteration." But I would rather not fiddle with the data structure if it is doing its job (as the record of current workflow state). I'm not too worried about backwards compatibility as such, but whether the proposed change is backwards compatible could be a good indication of how intrusive the change is.

**Comment by carolinedav** [comment]
Actually, I just realized that the `brer.log` file for each member tracks all iterations so this change is not necessary unless we wanted a `state.json` for each iteration.

**Comment by eirrgang** [comment]
Yes, all of the information should appear in the log file, but that isn't necessarily easy to parse programmatically. Did you just need a human-parseable record? Or do you think this information should be available programmatically (with API features or a nicely parseable file format)?

**Comment by carolinedav** [comment]
My main concern was a human-parseable record, but it would be nice to pull information programmatically! Like for instance, if I want to export general information like below into a csv or other data file for a comparison across members: 
```
{'general parameters': {'ensemble_num': 88, 'iteration': 0, 'phase': 'training', 'start_time': 0, 'A': 50, 'tau': 50, 'tolerance': 0.25, 'num_samples': 50, 'sample_period': 100, 'production_time': 10000}, 'pair parameters': {'107A-377A': {'sites': [9379, 13460], 'logging_filename': '107A-377A.log', 'alpha': 0.0, 'target': 3.0}, '107A-377B': {'sites': [9379, 5805], 'logging_filename': '107A-377B.log', 'alpha': 0.0, 'target': 3.0}, '107A-377C': {'sites': [9379, 21115], 'logging_filename': '107A-377C.log', 'alpha': 0.0, 'target': 3.0}, '107B-377A': {'sites': [1640, 13460], 'logging_filename': '107B-377A.log', 'alpha': 0.0, 'target': 3.0}, '107B-377B': {'sites': [1640, 5805], 'logging_filename': '107B-377B.log', 'alpha': 0.0, 'target': 3.0}, '107B-377C': {'sites': [1640, 21115], 'logging_filename': '107B-377C.log', 'alpha': 0.0, 'target': 3.0}, '107C-377A': {'sites': [16979, 13460], 'logging_filename': '107C-377A.log', 'alpha': 0.0, 'target': 3.0}, '107C-377B': {'sites': [16979, 5805], 'logging_filename': '107C-377B.log', 'alpha': 0.0, 'target': 3.0}, '107C-377C': {'sites': [16979, 21115], 'logging_filename': '107C-377C.log', 'alpha': 0.0, 'target': 3.0}}}
```

**Comment by eirrgang** [comment]
In that case, I think we should write one or two files per simulation phase. The code would look pretty much the same as the state output, but would probably occur in the same place as the `logger` calls. The existence of these files could actually be useful to provide some redundancy in determining whether a particular phase has begun and/or finished. (Thus potentially two files, if we want to maybe separately record input and output.)

My expectation is that it would make sense to write the files in the same directories as the simulation data so they could be as localized as possible and have static filenames (e.g. `input.json` and `results.json` or something). I _think_ (but have not confirmed) that we can essentially use the `state.json` machinery to write these files as sort of a serialization of the `RunConfig` (or `RunData`?) object. On that note, we might consider formally updating the `state.json` to be the artifact associated with a serializable/deserialize API aspect, but that might be a topic for the future.
